File: cliente_1/Protocol.cs
using System;

namespace ClienteChatCSharp
{
 
    public static class Protocol
    {


        public const string Login = "LOGIN";

        public const string Group = "GROUP";

        public const string Private = "PRIVATE";

        public const string List = "LIST";

        public const string Exit = "EXIT";


        public const string Welcome = "WELCOME";

        public const string Join = "JOIN";

        public const string Left = "LEFT";

        public const string Users = "USERS";

        public const string Error = "ERROR";

 
        public static string CrearLogin(string usuario)
        {
            return $"LOGIN|{usuario}";
        }

  
        public static string CrearMensajeGrupo(string mensaje)
        {
            return $"GROUP|{mensaje}";
        }


        public static string CrearMensajePrivado(
            string destinatario,
            string mensaje)
        {
            return $"PRIVATE|{destinatario}|{mensaje}";
        }

     
        public static string CrearSolicitudUsuarios()
        {
            return "LIST";
        }

        public static string CrearSalida()
        {
            return "EXIT";
        }
    }
}

File: cliente_2/protocol.py
class Protocol:
    """Comandos compatibles con el servidor Java del proyecto."""

    LOGIN = "LOGIN"
    GROUP = "GROUP"
    PRIVATE = "PRIVATE"
    LIST = "LIST"
    EXIT = "EXIT"

    WELCOME = "WELCOME"
    JOIN = "JOIN"
    LEFT = "LEFT"
    USERS = "USERS"
    ERROR = "ERROR"
    INFO = "INFO"

    @staticmethod
    def login(usuario):
        return f"LOGIN|{usuario}"

    @staticmethod
    def group(mensaje):
        return f"GROUP|{mensaje}"

    @staticmethod
    def private(destinatario, mensaje):
        return f"PRIVATE|{destinatario}|{mensaje}"

    @staticmethod
    def users():
        return "LIST"

    @staticmethod
    def exit():
        return "EXIT"
